src/Evolution/DiscontinuousGalerkin/SlopeLimiters/WenoGridHelpers.hpp

```diff
@@ -0,0 +1,57 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+
+/// \cond
+class DataVector;
+template <size_t>
+class Direction;
+template <size_t>
+class Mesh;
+/// \endcond
+
+namespace SlopeLimiters {
+namespace Weno_detail {
+
+// Compute the coordinate positions of a neighbor element's grid points in the
+// logical coordinates of the local element.
+//
+// However, the results are organized to match the expected input of a
+// RegularGridInterpolator constructor:
+// - The coordinates are given as an array of 1D coordinate values.
+// - For any dimension where the neighbor and local element share the same 1D
+//   mesh, the array is empty. (This tells the RegularGridInterpolator that no
+//   interpolation needs to be done in this dimension.)
+//
+// NOTE: Current implementation assumes, but does not check:
+// - No h-refinement, i.e., there is only one neighbor in each direction.
+// - No p-refinement, i.e., each element has the same mesh.
```

Comment:
where is this assumed?

Reply:
If the neighbor has a different set of grid points in the direction(s) perpendicular to the interface normal (i.e., parallel to the interface), there will need to be interpolation between the two transverse grids. It will be the job of this helper function to return the coordinates to interpolate to. This is totally ignored at the moment.

Reply:
Sorry, I don't see anything in this PR that depends upon the transverse dimensions, so either the comment needs to clarify how this assumption of the transverse meshes enters, or needs to be moved to whatever calls this function.   

Also is it assumed that the neighbor has an aligned OrientationMap?

Reply:
The reason you don't see anything using the transverse direction is because I assume it is the same in both elements. In the case that they are the same, no interpolation is needed in the transverse direction, and therefore no transverse coordinates are added to the `std::array` computed here.

It might also be easier to implement the p-refinement case than to document the assumptions being made... so maybe I'll just do that.

Reply:
Ah okay, I think I understand the p-refinement case now, and if so, I think it would be easier to just implement it.   

What about the orientation between elements?

Reply:
Re orientations: right now the code is indeed wrong (well spotted!), because a neighbor with the same mesh but in a different orientation will be handled as if the orientation was aligned. 

This will automatically be fixed when I implement p-refinement, because I will use the PackagedData from the limiter to obtain the neighbor mesh.... and there would be much bigger problems if the limiter doesn't correctly orient data when sending it :)